### DISCUSSION TITLE
delete button in design section

### DISCUSSION THREAD

**Discussion by juanesf** [meerk40t/meerk40t#1951] [Ideas]
Hello, Thank you very much for your software.
I am using a touch screen notebook and it is very comfortable to use with the software, but the design section is missing a delete button.

<img width="960" alt="image" src="https://github.com/meerk40t/meerk40t/assets/19717636/d86deb94-b28a-4dde-a55d-1483ea94e40d">

**Comment by tatarize** [comment]
Hm. Yeah, usually you'd use the delete key or rightclick the element in question, but you certainly have a point that delete/erase should be a tool or whatever or just give a button to delete some things.

**Reply by juanesf** [reply]
Exactly as mentioned

**Reply by tatarize** [reply]
So do you expect this delete button to let you click on things and delete them or just remove anything selected from the scene when you press it. I'm thinking the latter. There's an argument for an erase tool but that'd probably be more of a Node-Edit thing.

**Comment by tiger12506** [comment]
If they are using a touch screen that folds back on itself, then they dont have access to the keyboard, and need a delete button as a tool.
Seems reasonable thay when in select mode it would just delete, but when in node edit mode it could delete just points.

**Comment by tatarize** [comment]
I've added it to the main branch and to the bug fixes in 0.8.x
```python
        kernel.register(
            "button/tools/Delete",
            {
                "label": _("Delete"),
                "icon": icons8_delete_50,
                "tip": _(
                    "Delete selected items"
                ),
                "action": lambda v: kernel.elements("tree selected delete\n"),
                "size": bsize_normal,
            },
        )
```

In theory there may be a `clipboard cut` that could be used in a pinch.

**Reply by tatarize** [reply]
In fact, it's next to tools and shows up fine. It's not really delete and doesn't go to the real clipboard but it's at least there. I might actually should move delete to that panel.

**Reply by tatarize** [reply]
![image](https://github.com/meerk40t/meerk40t/assets/3302478/b59a344d-dc23-444c-a20a-1d7d0d9de0d2)

It basically does cutting without the clipboard, and is sort of right next to the clipboard thing but the implications there are radically different.

**Comment by juanesf** [comment]
I already downloaded version 0.8.x; It works correctly, I can record the operation if necessary.
Thanks again for your work.

https://we.tl/t-LcQjuxyzoq
